plugin/src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/rest/RestPPLQueryAction.java

```diff
@@ -44,17 +48,23 @@ public String getName() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
-        PPLService pplService = new PPLService();
-        return channel -> pplService.execute(new PPLQueryRequest(), new ResponseListener<PPLQueryResponse>() {
-            @Override
-            public void onResponse(PPLQueryResponse pplQueryResponse) {
-                channel.sendResponse(new BytesRestResponse(OK, "application/json; charset=UTF-8", "ok"));
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                channel.sendResponse(new BytesRestResponse(OK, "application/json; charset=UTF-8", "error"));
-            }
-        });
+        AnnotationConfigApplicationContext context = SecurityAccess.doPrivileged(
+                () -> new AnnotationConfigApplicationContext(
+                        PPLServiceConfig.class));
```

Comment:
Just curious, is this privilege call required for all classes annotated by Spring annotation?

Reply:
If we use Spring IoC, it should be fine. But for other features, I am not sure now.

Reply:
I see. Thanks!